Automattic/woocommerce-payments
# Fixes Pay for Order page for non-card payment methods

#### Changes proposed in this Pull Request


# Fixes Pay for Order page for non-card payment methods

In #8089, we attempted to prevent our gateway from intercepting other gateways when the order pay form is submitted. Unfortunately, this also prevented non-card payment methods from being used by our own plugin too. This PR provides a solution to this problem: instead of simply checking whether the `woocommerce_payments` gateway is selected (the card method), we check whether any WooPayments payment method is selected. 




#### Testing instructions


1. Create a new order and proceed to the Pay for Order page for the order.
2. Checkout for card and all non-card payment methods should succeed.
3. Install another payment gateway (e.g. Stripe) and ensure that this gateway also works on the Pay for Order page.



-------------------


**Post merge**
